File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    def check(x,y):
        target = (x+y)*2 - 4
        if target == brown:
            return True
        else:
            return False
    
    total = brown+yellow
    answerX, answerY = 0, 0
    for i in range(1,total):
        if total % i == 0:
            answerX = i
            answerY = total // i
        if(check(answerX, answerY)):
            answer.append(max(answerX,answerY))
            answer.append(min(answerX,answerY))
            return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        
        int total = brown + yellow;
        int sero = 0;
        int garo = 0;
        for(int i=1; i<total/2; i++) {
            if(total % i != 0) continue;
            sero = i;
            garo = total/i;
            if(checkBrownCount(sero, garo, brown))
                return new int[]{garo, sero};
        }
        return null;
    }
    
    private boolean checkBrownCount(int sero, int garo, int brown) {
        return ((garo + sero - 2) * 2) == brown;
    }
}
